player/lua/select.lua

```diff
@@ -0,0 +1,294 @@
+            or {"ffmpeg", "-loglevel", "quiet", "-i", mp.get_property("path"), "-map", "s:" .. sub["id"] - 1, "-f", "lrc", "-map_metadata", "-1", "-fflags", "+bitexact", "-"}
+    })
+
+    if r.status < 0 then
+        show_error("subprocess error: " .. r.error_string)
+        return
+    end
+
+    if r.status > 0 then
+        show_error("ffmpeg failed with code " .. r.status)
+        return
+    end
+
+    local sub_lines = {}
+    local default_item = 1
+
+    local sub_start = mp.get_property_native("sub-start", 0)
+    local m = math.floor(sub_start / 60)
+    local s = sub_start - m * 60
+    sub_start = string.format("%.2d:%05.2f", m, s)
+
+    -- Strip HTML and ASS tags.
+    for line in r.stdout:gsub("<.->", ""):gsub("{\\.-}", ""):gmatch("[^\n]+") do
+        sub_lines[#sub_lines + 1] = line
+
+        if line:find("^%[" .. sub_start) then
+            default_item = #sub_lines
+        end
+    end
+
+    input.select({
+        prompt = "Select a line to seek to:",
+        items = sub_lines,
+        default_item = default_item,
+        submit = function (index)
+            mp.commandv("seek", sub_lines[index]:match("[%d:%.]+"), "absolute")
+            input.terminate()
+        end,
+    })
+end)
+
+local function truncate(str)
+    return #str < 100 and str or str:sub(1, 99) .. "…"
+end
+
+mp.add_forced_key_binding(nil, "select-binding", function ()
+    local binding_map = {}
+
+    for _, binding in pairs(mp.get_property_native("input-bindings")) do
+        if binding.priority >= 0 and (
+               binding_map[binding.key] == nil or
+               (binding_map[binding.key].is_weak and not binding.is_weak) or
+               (binding.is_weak == binding_map[binding.key].is_weak and
+                binding.priority > binding_map[binding.key].priority)
+        ) then
+            binding_map[binding.key] = binding
+        end
+    end
+
+    local bindings = {}
+    for _, binding in pairs(binding_map) do
+        if binding.cmd ~= "ignore" then
+            bindings[#bindings + 1] = binding
+        end
+    end
+
+    table.sort(bindings, function (a, b)
+        return a.key < b.key
+    end)
+
+    local items = {}
+    for i, binding in ipairs(bindings) do
+        items[i] = truncate(binding.key .. " " .. binding.cmd)
+    end
+
+    input.select({
+        prompt = "Select a binding:",
+        items = items,
+        submit = function (i)
+            mp.command(bindings[i].cmd)
+            input.terminate()
+        end,
+    })
+end)
+
+mp.add_forced_key_binding(nil, "show-properties", function ()
+    local properties = {}
+
+    -- Don't log errors for renamed and removed properties.
+    local msg_level_backup = mp.get_property("msg-level")
+    mp.set_property("msg-level", msg_level_backup == "" and "cplayer=no"
+                                 or msg_level_backup .. ",cplayer=no")
+
+    for _, property in pairs(mp.get_property_native("property-list")) do
```

Comment:
It works nicely, but I would recursively print any tables also. Instead of doing
```
video-params: "{"pixelformat": ..., "w": ..., "h": ..., ...}"
```
I would unwrap it until we get to literal types.
```
video-params/pixelformat: ...
video-params/w: ...
video-params/h: ...
```

Reply:
Do you want to list only the subproperties or the parent property on top of the subproperties? If we only list subproperties printing properties when pressing return is no longer necessary.